kep/tbd-peaks-power-aware-scheduling/README.md

```diff
@@ -0,0 +1,326 @@
+
+With 'heterogeneous nodes' (variations in CPU architectures, a mixed cluster consisting of both VM and Bare Metal nodes, and dissimilar allocation of resources across nodes), cluster owners can reap benefits, as the power efficiency of different nodes may vary. This often necessitates the use of customized power models for each node within the cluster.
+
+#### Story 2
+
+Even with 'homogeneous nodes,' if the CPU vs. Power relationship is non-linear (such as piece-wise linear or concave), cluster owners can still benefit. Although the power efficiency might be consistent across all nodes, the varying CPU utilizations among nodes at any given time can significantly impact the aggregate power consumption of the cluster. Placing an incoming pod on one node versus another can influence this consumption.
+
+### Notes/Constraints/Caveats
+
+- The [Min, Max] score range for the plugin is user-configurable, defaulting to [0, 100].
+- Additionally, the plugin normalizes the generated scores within the supplied [Min, Max] range.
+- There isn't a singular model that comprehensively depicts the correlation between utilization and power consumption for every cluster node.
+- At any given time, the current utilization of a node can be identified through metrics data.
+- Determining the current power consumption can be achieved either through live metrics or by inference from the model that describes the relationship between node utilization and power consumption.
+- Retraining the model representing this relationship may be necessary to better align with the workloads running on the node.
+
+### Risks and Mitigations
+
+- The benchmark test below captures the overhead resulting from model inferencing latency, which appears to be negligible.
+- The accuracy of the model that captures the relationship between utilization and power relies heavily on the quality of the metrics used for its training. The current implementation supports integration with various user-selected metrics providers, such as node-metrics and Kepler
+- PEAKS plugin requires the power model parameters for each cluster node configured with it. However, the power model (utilization vs. power consumption) parameters are subjected to change (e.g., increase in the number of CPUs on a node results in a change in the power model) over time. To address this, `configmap` is used as a mechanism to pass the updated power model parameters of any cluster node seamlessly.
+
+## Design Details
+
+### PEAKS workflow
+![PEAKS workflow](./figs/PEAKS_workflow.png)
+
+Here is a brief discription of some of the steps in the above workflow:
+- Create a power model for each cluster node: If a power model suitable for the workload to be scheduled already exists, then the same can be used to avoid creating a new power model. Thus it is an optional step.
+  - PEAKS plugin allows the cluster owners to bring their own power models that best represent both the node behavior and the workload characteristics.
```

Comment:
Hi @Huang-Wei , looks like [Kubernetes doesn't support the concept of nodepool](https://stackoverflow.com/questions/74623525/why-there-is-no-concept-of-nodepool-in-kubernetes) as such while managed Kubernetes services like GKE/EKS/AKS support it. However, we have updated the description to reflect this information.

Reply:
yup, my idea was that a "virtual node pool" that uses label selector to group nodes. And each virtual node pool can map to one power model. Otherwise, for a 5k nodes, it doesn't make sense store data for _each_ node.

Reply:
Sure, agree with you. And we have already made the required update to address this comment. Moving this comment to resolved state.